lib/galaxy/main_config.py

```diff
@@ -24,17 +24,17 @@ def default_relative_config_paths_for(app_name: str) -> List[str]:
     return paths
 
 
-def absolute_config_path(path, galaxy_root):
-    if path and not os.path.isabs(path):
+def absolute_config_path(path, galaxy_root: Optional[str]) -> Optional[str]:
+    if path and not os.path.isabs(path) and galaxy_root:
         path = os.path.join(galaxy_root, path)
     return path
 
 
-def config_is_ini(config_file):
-    return config_file and (config_file.endswith(".ini") or config_file.endswith(".ini.sample"))
+def config_is_ini(config_file: Optional[str]) -> bool:
+    return bool(config_file and (config_file.endswith(".ini") or config_file.endswith(".ini.sample")))
```

Comment:
I don't think that's needed.

```suggestion
    return config_file and (config_file.endswith(".ini") or config_file.endswith(".ini.sample"))
```

Reply:
it's for mypy only. Unless we want a return value to have a type of `Union[None, str, Any]`.

Reply:
I stand corrected: if `config_file` is None or "", the expression is short-circuited and these would be possible return values without the `bool()`.